docs/docker_install.md

```diff
@@ -0,0 +1,150 @@
+condition. If you have to use an older version of docker, you also
+might have to run
+```bash
+  ln -sf ../run/systemd/resolve/resolv.conf /etc/resolv.conf # originally was ../run/systemd/resolve/stub-resolv.conf
+```
+This is necessary due to, on the one hand, systemd.resolved only
+listening to loopback address in a way which makes it unreachable from
+docker and, on the other hand, docker limitations which hardcodeds
+google DNS address with no option to configure custom DNS when it
+cannot re-use DNS config from /etc/resolv.conf as in this case.
+See [this](https://github.com/moby/moby/pull/37485)
+and [this](https://github.com/docker/libnetwork/issues/2068) link for
+background.
+
+## Get Started
+
+The easiest way to get started is to run
+```bash
+	make test
+```
+which will build your current locally committed branch and run the
+end-to-end tests based on a completely freshly setup sawtooth ledger
+in default configuration.  Starting from a good branch, this should
+also normally successfully run and hence allows also making sure that
+the docker setup is ok.
+
+## End-to-end testing
+
+You are also encouraged to run `make test` whenever you create a new pull
+request as it is easiest way to robustly check your commits. Doing so
+doesn't require any deeper understanding of docker or docker-compose
+and should run out-of the box.
+
+By default it will build and run tests only in SGX simulator mode but
+the makefile does honor
+the `SGX_MODE` environment variable if you want to test in SGX HW
+mode (in which case, though, you will have to put proper sgx attestation info
+files (`sgx_spid.txt`,  `sgx_spid_api_key`, `sgx_ias_key.pem`) into the
+sgx sub-directory or define the `PDO_SGX_KEY_ROOT` environment variable to
+point to a suiteable destination.
+See `../build/common-config.sh --help` for more information on
+`PDO_SGX_KEY_ROOT` and related settings.
+
+
+## Developing and manual testing with docker(-compose)
+
+If you want to run your own experiments in the test environment, you
+can run
+ ```bash
+make test-env-setup
+```
+which launches the various containers of the topology. You can then
+interact with the environment using the standard docker-compose commands like
+`exec` to log into individual containers, `ps` to see status, `top` to monitor
+cpu inside containers or `down` to shut the invironment down again.
+See [docker-compose commandline reference](https://docs.docker.com/compose/reference/)
+for more information on commands.  To make sure these commands run
+with settings consistent with on how you invoked `test-env-setup`, you
+can invoke them with also with the makefile target  `run` . Just
+invoke it as
+```bash
+	make run ARGS='<docker-compose subcommand & args>'
+```
+e.g.  run `make run ARGS='exec pdo-build bash'` to start a shell in
+the pdo-build container.
+
+If you define the `PDO_DEBUG_BUILD` environment variable, the make
+commands will (with help of [`sawtooth.debugging.yaml`](sawtooth.debugging.yaml)) build
```

Comment:
this link and quite a few below have to be redirected to ../docker

Reply:
i think i have all the links fixed. hard to debug md.

Reply:
FYI... fixed the reference... there is no sawtooth.debugging.yaml, it is sawtooth-pdo.debugging.yaml